workspaces/azureml_devtestlabs/porter.yaml

```diff
@@ -0,0 +1,122 @@
+name: tre-workspace-azureml-devtestlabs
+version: 0.1.0
+description: "A Azure TRE workspace with Azure Machine Learning and Dev Test Labs"
+registry: azuretre
+dockerfile: Dockerfile.tmpl
+required:
+  - docker
+
+credentials:
+  - name: azure_tenant_id
+    env: AZURE_TENANT_ID
+  - name: azure_subscription_id
+    env: AZURE_SUBSCRIPTION_ID
+  - name: azure_client_id
+    env: AZURE_CLIENT_ID
+  - name: azure_client_secret
+    env: AZURE_CLIENT_SECRET
+
+parameters:
+  - name: tre_id
+    type: string
+    description: "The ID of the parent TRE instance e.g., mytre-dev-3142"
+  - name: workspace_id
+    type: string
+    description: "Unique 4-digit workspace ID"
+  - name: location
+    type: string
+    description: "Azure location (region) to deploy to"
+  - name: address_space
+    type: string
+    description: "VNet address space for the workspace services"
+    default: "10.2.1.0/24"
+  - name: acr_name
+    type: string
+    description: "Name of the azure container registry"
+  - name: tfstate_resource_group_name
+    env: TF_VAR_tfstate_resource_group_name
+    type: string
+    description: "Resource group containing the Terraform state storage account"
+  - name: tfstate_storage_account_name
+    env:  TF_VAR_tfstate_storage_account_name
+    type: string
+    description: "The name of the Terraform state storage account"
+  - name: tfstate_container_name
+    env: TF_VAR_tfstate_container_name
+    type: string
+    default: "tfstate"
+    description: "The name of the Terraform state storage container"
+
+mixins:
+  - exec
+  - docker
+
+install:
+  - exec:
+      description: "docker login"
+      command: docker
+      arguments:
+        - login
+        - "{{ bundle.parameters.acr_name }}.azurecr.io"
+      flags:
+        u: "{{ bundle.credentials.azure_client_id }}"
+        p: "{{ bundle.credentials.azure_client_secret }}"
+  - exec:
+      description: "Install vanilla workspace"
+      command: porter
+      arguments:
+        - install
+      flags:
+        reference: "{{ bundle.parameters.acr_name }}.azurecr.io/tre-workspace-vanilla:v0.1.0"
+        cred: ./azure.json
+        parameter-set: ./parameters_workspace_vanilla.json
+        debug:
+  - exec:
+      description: "Install Azure ML Service"
+      command: porter
+      arguments:
+        - install
+      flags:
+        reference: "{{ bundle.parameters.acr_name }}.azurecr.io/tre-service-azureml:v0.1.0"
+        cred: ./azure.json
+        parameter-set: ./parameters_service_azureml.json
+        debug:
+
+upgrade:
+  - exec:
+      description: "Upgrade workspace"
+      command: echo
+      arguments:
+        - "This workspace does not implement upgrade action"
+
+# TODO: uninstall does not work at present, will need remote porter state
```

Comment:
I believe @tompaana has created a solution for remote state

Reply:
We have remote TF state, not remote porter state yet. Probably needs an issue. @tompaana ?

Reply:
I believe it does, if I try uninstall from a different porter install to that I installed from I get:

```
Error: 1 error occurred:
        * could not load installation tre-service-azureml: Installation does not exist
```        

I think we need to use:

https://porter.sh/plugins/azure/

"Storage plugins allow Porter to store data, such as **installation data** and credentials or parameters, in Azure cloud."

Reply:
The documentation is not clear :-/

I read it as it is an option to store credentials etc in Storage Account, but that does not explain why installing and uninstalling with two different Porter bundle instances does not work.

@tompaana / @jjcollinge do you know how the Azure Mixin works? Does it store state somewhere?

Reply:
Azure Plugin - not Azure Mixin - Mixins and Plugins are different. 

I tried to learn about the architecture: https://porter.sh/architecture/